src/API/PostService.ts

```diff
@@ -0,0 +1,28 @@
+import axios from "axios";
+import {IPostInList, IPostOnPage} from "../types/types";
+
+export default class PostService {
+    static async getAllFromPages(pages: number) {
+        if (pages === 0) {
+            return;
+        }
+        const array: IPostInList[][] = [];
+        for (let page = 1; page <= pages; page++) {
+            array.push(
+                (await (
+                    axios.get<IPostInList[]>(
+                        `https://api.hnpwa.com/v0/newest/${page}.json`
+                    )
+                )).data
+            );
+        }
+        // @ts-ignore
+        return [].concat(...array);
```

Comment:
Я так понял это ты хотел так ссылку новую сгенерить?
Лучше тогда так
```suggestion
        return [...array];
```
И ts ignore не нужен

Reply:
Ну тут не совсем так, я пытался превратить двумерный массив в одномерный, но уже нашел метод `flat()`, так что всё ок